src/BreakpointViewer.cpp

```diff
@@ -793,10 +817,42 @@ void BreakpointViewer::fillTableRow(int index, BreakpointRef::Type type, int row
 	auto* item5 = table->item(row, SEGMENT);
 	item5->setText(segment);
 
+	// id
 	auto* item6 = table->item(row, ID);
 	item6->setText(bp.id);
 }
 
+std::optional<Breakpoint> BreakpointViewer::parseTableRow(BreakpointRef::Type type, int row)
+{
+	Breakpoint bp;
+
+	auto* table = tables[type];
+	auto* model = table->model();
+	auto* combo = (QComboBox*) table->indexWidget(model->index(row, WP_TYPE));
+	bp.type = type == BreakpointRef::WATCHPOINT ? readComboBox(row) : Breakpoint::BREAKPOINT;
+
+	QString location = table->item(row, LOCATION)->text();
+	auto loc = parseLocationField(-1, type, location, combo ? combo->currentText() : "");
+	if (!loc) return {};
+	bp.address = loc->start;
+	bp.regionEnd = loc->end;
```

Comment:
Thanks for trying my suggestion to create a `AddressRange` and `Slot` struct. Now that I see it, I think it indeed looks better.
Next step (for a future patch?) could be to use these structs in more places. For example as fields in the `Breakpoint` struct. Or to replace 2 function parameters with a single one (e.g. in `createSetCommand()).